src/host/main.cpp

```diff
@@ -434,6 +434,11 @@ int main(int argc, char** argv)
       }
       LOG_INFO_FMT("Reading previous service identity from {}", idf);
       startup_config.recover.previous_service_identity = files::slurp(idf);
+      while (startup_config.recover.previous_service_identity->back() == 0x00)
+      {
+        // Trailing null byte not required, but tolerated.
```

Comment:
Is that because of 1.x service certificates?

Reply:
See response to Eddy's comment further up. I just added that to make sure we accept certs without the null byte, but I'm thinking about removing that commit as it looks more complicated to users, when they should just not worry about it.